spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/SparkFilters.java

```diff
@@ -168,9 +169,23 @@ public static Expression convert(Filter filter) {
 
         case NOT:
           Not notFilter = (Not) filter;
-          Expression child = convert(notFilter.child());
-          if (child != null) {
-            return not(child);
+          Filter childFilter = notFilter.child();
```

Comment:
We could try to implement `rewriteNot` for Spark filters but that sounds like a lot of work for such a small use case. We will have to create a custom `NotIn` Spark filter, a visitor to traverse Spark filters, a negation mapping, etc.

Reply:
why this is not fixed in the IN section ?